### PR TITLE
Add table aws_iot_greengrass_* Closes #1994

### LINKED ISSUE #1994: Add table aws_iot_*
Add new tables (plural!!)
- `aws_iot_core_things`
- `aws_iot_core_thing_groups`
- `aws_iot_core_thing_types`
- `aws_iot_core_fleet_metrics`
- `aws_iot_greengrass_devices`
- `aws_iot_greengrass_components`
- `aws_iot_greengrass_deployments`
- `aws_iot_greengrass_devices`
- `aws_iot_greengrass_groups`

**References**
[AWS IoT Core](https://aws.amazon.com/iot-core/?nc=sn&loc=2&dn=3)
[AWS IoT Greengrass](https://aws.amazon.com/greengrass/?nc=sn&loc=2&dn=2)

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select * from aws_iot_greengrass_deployment
+-----------------------+--------------------------------------+-------------------+----------------------+---------------------------+-------------------+-------------+------------------------------------------------------+----------------------------------------------->
| deployment_name       | deployment_id                        | deployment_status | is_latest_for_target | creation_timestamp        | parent_target_arn | revision_id | target_arn                                           | iot_job_arn                                   >
+-----------------------+--------------------------------------+-------------------+----------------------+---------------------------+-------------------+-------------+------------------------------------------------------+----------------------------------------------->
| Deployment for test12 | 40c2d4b5-1bcd-43da-b5ef-7199d1ade409 | ACTIVE            | true                 | 2023-12-06T18:21:35+05:30 | <null>            | 1           | arn:aws:iot:us-east-1:333333333333:thinggroup/test12 | arn:aws:iot:us-east-1:333333333333:job/8167384>
+-----------------------+--------------------------------------+-------------------+----------------------+---------------------------+-------------------+-------------+------------------------------------------------------+----------------------------------------------
```
</details>

### CODE REVIEW

**Comment by madhushreeray30** [comment]
@ParthaI Please also resolve the merge conflicts, thanks!

File: aws/table_aws_iot_greengrass_deployment.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+			}
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getIotGreengrassDeployment(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (khushboo9024) [inline aws/table_aws_iot_greengrass_deployment.go:195]: @ParthaI Please update the function name from `getIotGreengrassDeployment` to `getIoTGreengrassDeployment`

File: docs/tables/aws_iot_greengrass_deployment.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+select
+  deployment_name,
+  deployment_id,
+  is_latest_for_target,
+  creation_timestamp,
+  parent_target_arn,
+  target_arn
+from
+  aws_iot_greengrass_deployment;
+``
```
Comment (khushboo9024) [inline docs/tables/aws_iot_greengrass_deployment.md:43]:
```suggestion
```
```

**Review by khushboo9024 — COMMENTED** [review]
@ParthaI I am getting below error for table `aws_iot_greengrass_component`
```
> select * FROM aws_iot_greengrass_component
Error: aws: operation error GreengrassV2: GetComponent, https response error StatusCode: 400, RequestID: 5d4d4b3c-9b35-4d38-96b3-7d6a8888844a, ValidationException: Invalid Input: Request component version is blank (SQLSTATE HV000)
> 
```

Note: Also `GetConfig` is missing for this table, could you please take a look?

**Comment by ParthaI** [comment]
> Note: Also GetConfig is missing for this table, could you please take a look?

@khushboo9024  The `get config` is not included here for several reasons:

1. The API only returns the recipe and recipeOutputFormat, without any additional information.
2. A specific version is required to make the API call, but we obtain the latest version through the list call.
3. Utilizing the GetComponent API as a hydrated call is a more effective approach.
4. We can add a separate table table `aws_iot_greengrass_component_version` . We have a list/get API available for it.

**Comment by misraved** [comment]
We are closing this PR since it is not mapped to current priorities. We will reopen this PR once there is a need to implement this table.

**Pull request closed without merging**